xarray/tutorial.py

```diff
@@ -6,23 +6,78 @@
 
 """
 import hashlib
-import os as _os
-from urllib.request import urlretrieve
+import pathlib
+import shutil
+import tempfile
 
 import numpy as np
+import requests
```

Comment:
don't we need to add `requests` to a list of dependencies somewhere?

Reply:
yes, that's what I was thinking, too. Should that be a mandatory or an optional dependency?

Reply:
i think it's nice to have tutorial functions that always work. So i guess a mandatory dependency.

cc @pydata/xarray

Reply:
+1 for a mandatory dependency, just so it gets installed by default. I think it's fine to add a hard dependency on `requests`. It's already distributed on nearly every platform, and if it's hard to install for some reason it's easy for a redistributer to simply drop `xarray.tutorial`.

Reply:
done